### PR TITLE
LPS-198941 search-experiences-rest-impl: Use the default language from when sxpElement was created

### PR DESCRIPTION
**Pull request by gustavolimav**
https://liferay.atlassian.net/browse/LPS-198941

### CODE REVIEW

File: modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/dto/v1_0/converter/SXPElementDTOConverter.java [hunk 1]
```diff
@@ -59,12 +59,14 @@ public SXPElement toDTO(
 			com.liferay.search.experiences.model.SXPElement sxpElement)
 		throws Exception {
 
+		String defaultLanguageId = sxpElement.getDefaultLanguageId();
```
Comment (joshuacords) [inline modules/dxp/apps/search-experiences/search-experiences-rest-impl/src/main/java/com/liferay/search/experiences/rest/internal/dto/v1_0/converter/SXPElementDTOConverter.java:62]: Hi @gustavolimav, could you inline this like the dtoConverterContext.getLocale() was?

**Comment by gustavolimav** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
Hi @gustavolimav, so I tested this scenario:

1. Startup a bundle before these changes (Title & Descriptions are blank)
2. Shutdown and deploy these changes
3. Restart and view the Elements: Title & Descriptions are still blank, this is because sxpElement.getDefaultLanguageId() will return pt_BR.

**Comment by joshuacords** [comment]
Hi @gustavolimav, 

You can take a look at my attempt here: https://github.com/joshuacords/liferay-portal/commits/LPS-198941-2

I can't say it's pretty but it does resolve the title and description problems for pre-created elements. But when I copy a precreated element I get empty elements - I think this is because the Localization Maps for the Title and Description are empty. I attempted to put something in the map, but it failed and I don't really like the direction it's going.

**Comment by gustavolimav** [comment]
@joshuacords didn't have time to look into your solution but I did one and committed here, can you please take a look at it?
I will only be able to look again tomorrow

**Comment by joshuacords** [comment]
Hi @gustavolimav, I tested that but the old versions were still blank. The getDescriptionMap returns an empty map because only [available locales are extracted](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/portal-impl/src/com/liferay/portal/util/LocalizationImpl.java#L506-L522).

Considering another option, we could look to make an upgrade process, however I believe switching locales on the site would probably still break the elements.

**Comment by gustavolimav** [comment]
Hey, @joshuacords, I tested it again and it seems to have worked here 😕 ❓ 

I believe the best approach would be to make an upgrade, so we can make ootb Elements and Custom Elements have the same behavior regarding defaultLanguageId, where we could get both using the defaultLanguageId and not using the hardcoded en_US locale to ootb.

**Comment by gustavolimav** [comment]
Hey @joshuacords added a commit with some idea of how the upgrade could be done

**Comment by joshuacords** [comment]
I just want to leave a note on my opinions with the fix:

1. We need to make sure that customers that have upgraded not using the en_US locale will have no problems with the default elements that were created in earlier versions. That's why 2190b419ee5b1bb3780a4ad893db7e587323870c helps fix it going forward but isn't sufficient in itself.
2. The shortcoming of the upgrade is there's still the potential for the customer to create a site in a default language, and then remove that language later. (They create a site without portal-properties which uses the en_US and then later restrict the site to so other language.) I believe in this case the Element Titles & Descriptions will be empty again.
3. If we do go with the upgrade, we probably should only upgrade the default Elements (I believe we did this before for adding the ERC code) not all Elements.
4. I think I like the Fallback option the best, as it should deal with everything, however we need to make sure it covers a multitude of cases: Copy/Export, Locale Changes, and that it works with previously created Elements.

**Pull request closed without merging**